OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/internal/permissions/impl/NotificationPermissionController.kt

```diff
@@ -64,14 +70,35 @@ internal class NotificationPermissionController(
             )!!
 
     init {
+        this.enabled = notificationsEnabled()
```

Comment:
Not sure if `_application.appContext` is set at this point, `notificationsEnabled()` will always return `false` if it is not.

Reply:
It has always been set and returns the correct value in my testing, but that is not to say that it couldn't happen. If it isn't then we will just fire the handler unnecessarily which isn't the worst thing. We could also save the value in sharedPreferences to reduce the likely hood of this occurring.

Reply:
I wasn't actually sure off hand so wanted to make sure we know if this was the case / tested. Since it is working I think it's fine to set it like this.
* However I can see this being fragile, but caching it has it own problem, as it could be out of date.